### ISSUE TITLE
Can SVG icons have fill:currentColor by default?

### ISSUE THREAD

**Issue by arminrosu** [carbon-design-system/carbon#5531]
## What package(s) are you using?

- [x] `@carbon/icons-react`
- [x] `carbon-components`
- [x] `carbon-components-react`

## Summary

Hello. I am wondering why SVG icons don't have `fill: currentColor` by default. Thus, it will inherit the `color` property of its container. This simplifies the [process of setting the color](https://github.com/carbon-design-system/carbon/tree/master/packages/icons-react#icon-fill) a bit.

Of course, I can create a HOC / wrapper for the icons where I do this myself.

I was just curious if you had any reasons for not having it and what those might be.

**Comment by joshblack** [comment]
Definitely think this is a great idea! Would not be opposed to it 😄 I doubt there would be too many ramifications to setting this, right? The only thing would be how to ship it such that class names being passed in don't get overruled (as compared to inline styles)

**Comment by arminrosu** [comment]
Awesome :)

I wrote a small sample to check how `fill: currentColor;` would interact with the SVGs: https://codepen.io/arminrosu/pen/QWbObVr

SVGs are globally set to have a fill of `currentColor`: 

```css
svg {
  fill: currentColor;
}
```

Interestingly, `<svg><use href="#embedded_SVG_ID" ></use></svg` cannot have it's `color` overridden by `fill` (red in the [URL above](https://codepen.io/arminrosu/pen/QWbObVr))

Embedded SVGs do not have that problem (example 3). Attribute selection is also possible, for duo-tone icons.

**TLDR;**

Adding `svg { fill: currentColor; }` to the library will not affect current functionality, but modify all icons to have the text color by default.

What do you think?

**Comment by tw15egan** [comment]
This would be nice, ~the only thing I can think of is that disabled states may not always have an updated color when disabled, so we'd just need to look out for that~ 

Took a look around and looks like we do set the color when disabled, so it shouldn't be an issue

**Comment by joshblack** [comment]
Closed by: https://github.com/carbon-design-system/carbon/pull/6179 👀

**Comment by SimonFinney** [comment]
This appears to cause regression in dark themes in cases where the icon is dependent on a `fill` from its parent - here's an example of the `Search` component:

![Default](https://user-images.githubusercontent.com/3846874/87023869-4a2c5880-c1d0-11ea-8d9e-3bda3c07415b.png)

This is resolved by removing the `currentColor` on the icon as per the original behavior: 

![currentColor removed](https://user-images.githubusercontent.com/3846874/87024141-a3948780-c1d0-11ea-9650-aa540b21fa8c.png)

But also if the `color` of the parent is set to the `fill` value:

![color](https://user-images.githubusercontent.com/3846874/87024521-23225680-c1d1-11ea-8986-c13b738d7271.png)

We can use this as a workaround for now, but I wanted to raise it for awareness as I suspect this may be a silent issue reoccurring elsewhere and could easily be overlooked on lighter themes.

**Comment by joshblack** [comment]
Thanks @SimonFinney! Opened an issue over at: https://github.com/carbon-design-system/carbon/issues/6432
